pkg/plugins/plugins.go

```diff
@@ -0,0 +1,168 @@
+package plugins
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/openshift/console/pkg/serverutils"
+	"k8s.io/klog"
+)
+
+type PluginsHandler struct {
```

Comment:
this could just as well be a controller that holds a sync-lock-guarded map with enabled plugins and updates it accordingly to events from the consoleplugins informer

Reply:
advantage: no need to restart when a new plugin appears

Reply:
we have to restart. We are passing the array of the enabled ConsolePlugins into console by `console-config.yaml`. For the console to pick up the changes it has to be redeployed.
Thats the current state the ConsolePlugins functionality will work as declared in the [enhancement](https://github.com/openshift/enhancements/pull/441) doc.